pkg/loop/internal/relayer/pluginprovider/contractreader/contract_reader.go

```diff
@@ -121,6 +134,22 @@ func DecodeVersionedBytes(res any, vData *pb.VersionedBytes) error {
 			return fmt.Errorf("%w: %w", types.ErrInternal, err)
 		}
 		err = dec.Unmarshal(vData.Data, res)
+	case ValuesEncodingVersion:
+		protoValue := &valuespb.Value{}
+		err = proto.Unmarshal(vData.Data, protoValue)
+		if err != nil {
+			return fmt.Errorf("%w: %w", types.ErrInvalidType, err)
+		}
+
+		valuePtr, ok := res.(*values.Value)
+		if !ok {
+			return fmt.Errorf("%w: result is not of Value type", types.ErrInvalidType)
+		}
+
+		*valuePtr, err = values.FromProto(protoValue)
+		if err != nil {
+			return fmt.Errorf("%w: %w", types.ErrInvalidType, err)
+		}
```

Comment:
This part feels a bit wrong.  If the return value somehow isn't a values.Value, you should unwrap the value.Value to it.

Reply:
done